pkg/ebpf/telemetry/errors_collector_linux.go

```diff
@@ -8,112 +8,45 @@
 package telemetry
 
 import (
-	"errors"
-	"fmt"
-	"hash"
-	"hash/fnv"
-	"sync"
-	"syscall"
-	"unsafe"
-
-	manager "github.com/DataDog/ebpf-manager"
-	"github.com/cilium/ebpf"
-	"github.com/cilium/ebpf/asm"
-	"github.com/prometheus/client_golang/prometheus"
-	"golang.org/x/exp/slices"
-	"golang.org/x/sys/unix"
-
-	"github.com/DataDog/datadog-agent/pkg/network/ebpf/probes"
-	"github.com/DataDog/datadog-agent/pkg/util/kernel"
 	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/prometheus/client_golang/prometheus"
+	"unsafe"
 )
 
-const (
-	maxErrno    = 64
-	maxErrnoStr = "other"
-
-	ebpfMapTelemetryNS    = "ebpf_maps"
-	ebpfHelperTelemetryNS = "ebpf_helpers"
-)
-
-const (
-	readIndx int = iota
-	readUserIndx
-	readKernelIndx
-	skbLoadBytes
-	perfEventOutput
-)
-
-var ebpfMapOpsErrorsGauge = prometheus.NewDesc(fmt.Sprintf("%s__errors", ebpfMapTelemetryNS), "Failures of map operations for a specific ebpf map reported per error.", []string{"map_name", "error"}, nil)
-var ebpfHelperErrorsGauge = prometheus.NewDesc(fmt.Sprintf("%s__errors", ebpfHelperTelemetryNS), "Failures of bpf helper operations reported per helper per error for each probe.", []string{"helper", "probe_name", "error"}, nil)
-
-var helperNames = map[int]string{
-	readIndx:        "bpf_probe_read",
-	readUserIndx:    "bpf_probe_read_user",
-	readKernelIndx:  "bpf_probe_read_kernel",
-	skbLoadBytes:    "bpf_skb_load_bytes",
-	perfEventOutput: "bpf_perf_event_output",
+// EbpfErrorsCollector implements the prometheus Collector interface
+// for collecting statistics about errors of ebpf helpers and ebpf maps operations.
+type EbpfErrorsCollector struct {
```

Comment:
inconsistent caps, suggest `EBPFErrorsCollector` and `NewEBPFErrorsCollector()` because Go uses uppercase for abbreviations.